lib/sdl/main_sdl.cpp

```diff
@@ -397,24 +390,98 @@ void wzDisplayDialog(DialogType type, const char *title, const char *message)
 	SDL_ShowSimpleMessageBox(sdl_messagebox_flags, title, message, WZwindow);
 }
 
-SDL_WindowFlags getSDLFullscreenMode()
+WINDOW_MODE wzGetCurrentWindowMode()
 {
-	return WZ_SDL_FULLSCREEN_MODE;
-}
+	if (!WZbackend.has_value())
+	{
+		// return a dummy value
+		return WINDOW_MODE::windowed;
+	}
+	ASSERT(WZwindow != nullptr, "window is null");
 
-void wzToggleFullscreen()
-{
 	Uint32 flags = SDL_GetWindowFlags(WZwindow);
-	if (flags & getSDLFullscreenMode())
+	if (flags & SDL_WINDOW_FULLSCREEN_DESKTOP)
```

Comment:
```suggestion
	if ((flags & SDL_WINDOW_FULLSCREEN_DESKTOP) == SDL_WINDOW_FULLSCREEN_DESKTOP)
```
`SDL_WINDOW_FULLSCREEN_DESKTOP` is (oddly) defined as `SDL_WINDOW_FULLSCREEN | 0x00001000`:
https://github.com/spurious/SDL-mirror/blob/99dc02a820eea528b992964769430f2e44c5983f/include/SDL_video.h#L110